src/ast/passes/fold_literals.cpp

```diff
@@ -0,0 +1,450 @@
+#include "ast/passes/fold_literals.h"
+#include "ast/ast.h"
+#include "ast/visitor.h"
+#include "bpftrace.h"
+#include "log.h"
+#include "util/int_parser.h"
+
+namespace bpftrace::ast {
+
+namespace {
+
+class LiteralFolder : public Visitor<LiteralFolder, Expression *> {
```

Comment:
I feel like there should be a ton of unit tests for this class. Any reason not to have them?

Reply:
No, I will add them to this change shortly. I just want to get the architecture bits for the stack out of the way.

Reply:
I've added a bunch of tests around this. They look like the other tests for now, for simplicity.

Reply:
The tests look good!